SAMLPortal/Controllers/SetupController.cs

```diff
@@ -0,0 +1,268 @@
+using System.Linq;
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using SAMLPortal.Misc;
+using SAMLPortal.Models;
+using SAMLPortal.Models.Setup;
+using MySql.Data.MySqlClient;
+using Microsoft.EntityFrameworkCore;
+using CountryData.Standard;
+using Novell.Directory.Ldap;
+
+namespace SAMLPortal.Controllers
+{
+	[Route("Setup")]
+	[ServiceFilter(typeof(SetupAsyncActionFilter))]
+	public class SetupController : Controller
+	{
+		[HttpGet]
+		[Route("0")]
+		[Route("")]
+		public IActionResult Index()
+		{
+			return View();
+		}
+
+		[HttpGet]
+		[Route("1")]
+		public IActionResult StepOne()
+		{
+			var mysqlHost = Environment.GetEnvironmentVariable("SP_MYSQL_HOST");
+			var mysqlPortString = Environment.GetEnvironmentVariable("SP_MYSQL_PORT");
+			var mysqlDb = Environment.GetEnvironmentVariable("SP_MYSQL_DB");
+			var mysqlUser = Environment.GetEnvironmentVariable("SP_MYSQL_USER");
+
+			var mysqlPort = 3306;
+			if (mysqlPortString != null)
+			{
+				try
+				{
+					mysqlPort = Convert.ToInt32(mysqlPort);
+				}
+				catch (Exception)
+				{
+				}
+			}
+
+			StepOneModel model = new StepOneModel();
+			model.MySQLHost = mysqlHost != null ? mysqlHost : "";
+			model.MySQLPort = mysqlPort != 3306 ? mysqlPort : 3306;
+			model.MySQLDatabaseName = mysqlDb != null ? mysqlDb : "";
+			model.MySQLUser = mysqlUser != null ? mysqlUser : "";
+
+			return View(model);
+		}
+
+		[HttpPost]
+		[Route("1")]
+		public IActionResult StepOne(StepOneModel model)
+		{
+			if (ModelState.IsValid)
+			{
+				try
+				{
+					string connectionString = "server=" + model.MySQLHost
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'model' before using it.](https://app.codacy.com/manual/zegorax/SAMLPortal/pullRequest?prid=5343548)